catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -817,4 +797,26 @@ protected void validateGetResourceRequest(ResourceRequest getResourceRequest)
           "Cannot perform getResource with null attribute value or null attributeName, either passed in from endpoint, or as output from PreResourcePlugin");
     }
   }
+
+  private URI getDerivedUri(Metacard metacard, String qualifier) {
+    List<Serializable> derivedUris = metacard.getAttribute(Core.DERIVED_RESOURCE_URI).getValues();
+    URI resourceUri = null;
+
+    for (Serializable ser : derivedUris) {
+      try {
+        if (ser instanceof String) {
+          URI uri = new URI((String) ser);
+
+          if (uri.getFragment().equals(qualifier)) {
+            resourceUri = uri;
+            return resourceUri;
+          }
+        }
+      } catch (URISyntaxException e) {
+        LOGGER.debug("A metacard has a derived uri without the proper syntax.");
```

Comment:
I assume derived uris are checked for the right format when they're saved so this could just be a no-op but if not then I should do something more attention grabbing than a debug log. Like maybe throw an UnsupportedResourceException or something?

Reply:
I think I'll just make it an error level message and include the url.